src/lib/analytics.ts

```diff
@@ -0,0 +1,49 @@
+import AnalyticsSegment from 'analytics-node';
+
+import { Logger } from '@nestjs/common';
+
+export enum AnalyticsTrackName {
+  ShareLinkViewed = 'Share Link Viewed',
+  ShareLinkCopied = 'Share Link Copied',
+}
+
+export default class Analytics {
+  analytics: AnalyticsSegment;
+  logger: Logger;
+  static instance: Analytics;
+  constructor() {
+    this.logger = new Logger();
+    this.analytics = new AnalyticsSegment(process.env.APP_SEGMENT_KEY || 'xxx');
```

Comment:
avoid defaults for critical values (often related to environment variables) please